dhis-2/dhis-api/src/main/java/org/hisp/dhis/scheduling/JobProgress.java

```diff
@@ -110,11 +113,26 @@ public interface JobProgress {
    * Flow Control API:
    */
 
+  /**
+   * OBS! Should only be called after the task is complete and no further progress is tracked.
+   *
+   * @return true, if all processes in this tracking object were successful.
+   */
+  default boolean isSuccessful() {
```

Comment:
is having these default methods increasing the risk of bugs? It relies on the implementer overriding these default method and providing their own logic (they don't have to and could call these methods, giving unexpected results). Would it not be better to force the implementer to override by leaving it abstract?

Reply:
maybe we can chat about this when giving the demo next week, it's still a bit unclear for me :)